### PR TITLE
[BEAM-9160] Update AWS SDK to support Pod Level Identity

### PR DESCRIPTION
**Pull request by andeb**
Upgraded AWS SDK and added support to [pod identity](https://aws.amazon.com/blogs/opensource/introducing-fine-grained-iam-roles-service-accounts/) when running in AWS EKS.

------------------------------------------------------------------------------------------------

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Samza | Spark
--- | --- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_VR_Flink/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_VR_Spark/lastCompletedBuild/)
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Batch/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Streaming/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Spark_Batch/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_SparkStructuredStreaming/lastCompletedBuild/)
Python | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python2/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python35/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python36/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python37/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python2_PVR_Flink_Cron/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python35_VR_Flink/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_VR_Spark/lastCompletedBuild/)
XLang | --- | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_XVR_Flink/lastCompletedBuild/) | --- | --- | ---

Pre-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

--- |Java | Python | Go | Website
--- | --- | --- | --- | ---
Non-portable | [Build Status](https://builds.apache.org/job/beam_PreCommit_Java_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python_Cron/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PreCommit_PythonLint_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Go_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Website_Cron/lastCompletedBuild/) 
Portable | --- | [Build Status](https://builds.apache.org/job/beam_PreCommit_Portable_Python_Cron/lastCompletedBuild/) | --- | ---

See [.test-infra/jenkins/README](https://github.com/apache/beam/blob/master/.test-infra/jenkins/README.md) for trigger phrase, status and link of all Jenkins jobs.

### CODE REVIEW

**Comment by andeb** [comment]
R: @aaltay @kennknowles

**Comment by andeb** [comment]
R: @jbonofre @lukecwik @chamikaramj @timrobertson100

**Comment by lukecwik** [comment]
Can we run the linkage checker to perform analysis on whether we increased the number of linkage errors?

Example PR of running the linkage checker:
https://github.com/apache/beam/pull/10769

**Comment by lukecwik** [comment]
The code changes look fine otherwise.

**Comment by iemejia** [comment]
I did a run of @suztomo's awesome script and it seems amazon-web-services2 is introducing a lot of unrelated graalvm crap YAY !
https://gist.github.com/iemejia/a9aa0c1d8f4a0ec22867c3df5471e1d3

**Comment by iemejia** [comment]
Merged manually to remove some extra chars in the commit title.
Thanks again @andeb !

File: sdks/java/io/amazon-web-services/src/main/java/org/apache/beam/sdk/io/aws/options/AwsModule.java [hunk 1]
```diff
@@ -122,6 +123,8 @@ public AWSCredentialsProvider deserializeWithType(
         return new SystemPropertiesCredentialsProvider();
       } else if (typeName.equals(ProfileCredentialsProvider.class.getSimpleName())) {
         return new ProfileCredentialsProvider();
+      } else if (typeName.equals(WebIdentityTokenCredentialsProvider.class.getSimpleName())) {
+        return WebIdentityTokenCredentialsProvider.create();
```
Comment (ecapoccia) [inline sdks/java/io/amazon-web-services/src/main/java/org/apache/beam/sdk/io/aws/options/AwsModule.java]: Hi, interestingly I wrote a similar code to this to upgrade AWS client and support Pod Identity.

However I did not have to add another "else if" case; I exploited the existing DefaultAWSCredentialsProviderChain which you can call by specifying "--awsCredentialsProvider={\"@type\": \"DefaultAWSCredentialsProviderChain\"}".

The default provider chain already incorporates the WebIdentityTokenCredentialsProvider.

See https://github.com/aws/aws-sdk-java/blob/master/aws-java-sdk-core/src/main/java/com/amazonaws/auth/DefaultAWSCredentialsProviderChain.java#L48 here for the relevant line in the AWS client.

Comment (iemejia) [reply sdks/java/io/amazon-web-services/src/main/java/org/apache/beam/sdk/io/aws/options/AwsModule.java]: Nice info, @andeb would you be up to do the minor cleanup?

Comment (ecapoccia) [reply sdks/java/io/amazon-web-services/src/main/java/org/apache/beam/sdk/io/aws/options/AwsModule.java]: yup will do @iemejia

**Comment by andeb** [comment]
Thanks for reviewing and merging it, @iemejia!

With regards to the linkage errors, they may be false positives considering that there are lines of code to ignore classes from GraalVm? https://github.com/GoogleCloudPlatform/cloud-opensource-java/blob/master/dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/LinkageChecker.java#L208-L213

Also, I thought it made sense to expose `WebIdentityTokenCredentialsProvider` directly to be consistent with other exposed credential providers. Happy to do any cleanup though but it seems that @ecapoccia is already on it!

**Comment by suztomo** [comment]
The linkage errors from svm is false positives.

File: sdks/java/io/amazon-web-services/src/main/java/org/apache/beam/sdk/io/aws/options/AwsModule.java (hunk 1 above)
Comment (ecapoccia) [reply sdks/java/io/amazon-web-services/src/main/java/org/apache/beam/sdk/io/aws/options/AwsModule.java]: @iemejia @andeb opened PR https://github.com/apache/beam/pull/10836 as discussed.

**Pull request closed without merging**